server/src/services/patient_service.ts

```diff
@@ -0,0 +1,38 @@
+import "reflect-metadata";
+import { inject, injectable, named } from "inversify";
+import { StatusRepository } from "../repositories/status_repository";
+import { StatusFields } from "../entities/status_fields";
+import { PatientRepository } from "../repositories/patient_repository";
+import { AuthorizationError } from "../entities/errors/authorization_error";
+
+@injectable()
+export class PatientService {
+    constructor(
+        @inject("Repository")
+        @named("PatientRepository")
+        private readonly patientRepository: PatientRepository,
+        @inject("Repository")
+        @named("StatusRepository")
+        private readonly statusRepository: StatusRepository,
+    ) {}
+
+    async setStatusFields(doctorId: number, patientId: number, fields: StatusFields): Promise<void> {
+        // verify 3 required fields are present and true
+        const includesRequiredFields = !!(fields.temperature && fields.weight && fields.otherSymptoms);
```

Comment:
why the !!? i'm assuming it does more than negate twice?

Reply:
JS / TS use truthy and falsy values and doesn't get automatically cast to booleans. For example if any of these are undefined the statement will resolve to undefined without the !!

So this way I'm doing an explicit type conversion instead of an implicit one during the evaluation.

Using !! or Boolean() are the same thing just a preference.